RUMM-264 Introduce the Logs Sampler (#188)
DataDog/dd-sdk-android: docs/log_collection.md

| `setServiceName(<SERVICE_NAME>)` | Set `<SERVICE_NAME>` as value for the `service` [standard attribute][4] attached to all logs sent to Datadog.                                                                                                                                                           |
| `setLogcatLogsEnabled(true)`     | Set to `true` to use Logcat as logger.                                                                                                                                                                                                                                  |
| `setDatadogLogsEnabled(true)`    | Set to `true` to send logs to Datadog.                                                                                                                                                                                                                                  |

> | `setBundleWithTraceEnabled(true)`| Set to `true` to bundle the logs with the active Trace in your application. This parameter is true by default and it will allow to see all the logs sent during a specific Trace by using the Datadog Dashboard.                                                        |

Reviewer: ```suggestion
| `setBundleWithTraceEnabled(true)`| Set to `true` (default) to bundle the logs with the active trace in your application. This parameter shows all the logs sent during a specific trace by using the Datadog dashboard.                                                        |
```

| `setLoggerName(<LOGGER_NAME>)`   | Set `<LOGGER_NAME>` as the value for the `logger.name` attribute attached to all logs sent to Datadog.                                                                                                                                                                  |

> | `setVerbosity(Log.INFO)`         | Set the verbosity of the logger. All internal messages in the library with a priority equal or higher than the provided level will be logged to Android's Logcat.                                                                                                       |

Reviewer: ```suggestion
| `setVerbosity(Log.INFO)`         | Set the verbosity of the logger. All internal messages in the library with a priority equal to or higher than the provided level are logged to Android's Logcat.                                                                                                       |
```

> | `setSampleRate(<SAMPLE_RATE>)`   | Set the sampling rate for this logger. All the logs produced by this Logger instance will be randomly sampled according with the provide sample rate (default 1.0 meaning that all logs will be sent). Please note that the Logcat logs will not be sampled.            |

Reviewer: ```suggestion
| `setSampleRate(<SAMPLE_RATE>)`   | Set the sampling rate for this logger. All the logs produced by the logger instance are randomly sampled according to the provided sample rate (default 1.0 = all logs). **Note**: The Logcat logs are not sampled.            |
```
